contrib/internal/validationtest/validation_test.go

```diff
@@ -0,0 +1,198 @@
+package validationtest
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	memcachetest "gopkg.in/DataDog/dd-trace-go.v1/contrib/internal/validationtest/integrations/gomemcache/memcache"
+	dnstest "gopkg.in/DataDog/dd-trace-go.v1/contrib/internal/validationtest/integrations/miekg/dns"
```

Comment:
Doing this results in a `use of internal package ... not allowed to compile` error when trying to import the tests within the validationtest file.

Reply:
How about putting the test utils in `contrib/<path>/<to>/<dir>/internal/test` instead of `contrib/<path>/<to>/<dir>/internal` ?

Reply:
Bumping this thread as I believe we can avoid re-creating individual and long contrib folders.
e.g `contrib/internal/validationtest/contrib/miekg/dns/integration.go` should be moved to `contrib/miekg/dns/internal/test/integration.go`. The `contrib/miekg/dns` dir already exists.

Reply:
The problem is we cannot import an internal sub-package from here since the Go compiler doesn't allow so:

> When the go command sees an import of a package with internal in its path, it verifies that the package doing the import is within the tree rooted at the parent of the internal directory. For example, a package .../a/b/c/internal/d/e/f can be imported only by code in the directory tree rooted at .../a/b/c. It cannot be imported by code in .../a/b/g or in any other repository.

So in this case we have:

`contrib/internal/validationtest`

And the proposed internal subpackages would be:

`contrib/miekg/dns/internal/test`

So since the only common root from both directories is `contrib/`, this wouldn't be possible and any shared internal code needs to be placed under `contrib/internal` (sadly for us 😢 ).

Reply:
I think an alternative to have shorter paths would be to have a big package `contrib/internal/<some-name>`, where each integration is in its own file, like `miekg_dns.go`, `net_http.go`, etc.

Reply:
An extra alternative would be to have these sub-packages not being internal, like  `contrib/miekg/dns/test` - not sure how we feel about this approach...

Reply:
Thanks for the explanation @rarguelloF !

> I think an alternative to have shorter paths would be to have a big package contrib/internal/<some-name>, where each integration is in its own file, like miekg_dns.go, net_http.go, etc.

I like the idea! We can do it this way for now and move files to separate packages per integration in the future if necessary.

Reply:
miekg_dns.go, net_http.go, etc can live under `contrib/internal/validationtest` or even ``contrib/internal/validationtest/contribs`, right?

Reply:
Since we redeclare the `Integration` struct for each contrib test file, they need to have separate packages and therefore live in separate folders, otherwise I was getting errors trying to put everything in the same directory. I changed the structure to be: `contrib/internal/validationtest/miekg/dns.go`, `contrib/internal/validationtest/bradfitz/memcache.go` ... etc

Reply:
@wconti27 if we follow the big package approach, each struct needs to be named differently as they share the same namespace. e.g. `NetHTTP`, etc.